msteams-platform/toolkit/develop-single-sign-on-experience-in-Teams.md

```diff
@@ -0,0 +1,468 @@
+          clientSecret: SECRET_AAD_APP_CLIENT_SECRET
+          objectId: AAD_APP_OBJECT_ID
+          tenantId: AAD_APP_TENANT_ID
+          authority: AAD_APP_OAUTH_AUTHORITY
+    ```
+
+    > [!NOTE]
+    > Replace the `name` value with the desired name for your Microsoft Teams app.
+
+* Add the following code `aadApp/update` under `provision` in `./teamsapp.yml` and `./teamsapp.local.yml` to update your Microsoft Entra app. For more information, see [`aadApp/update`](https://github.com/OfficeDev/teams-toolkit/wiki/Available-actions-in-Teams-Toolkit#aadappupdate):
+
+    ```yaml
+    - uses: aadApp/update
+      with:
+        manifestPath: "./aad.manifest.json"
+        outputFilePath: "./build/aad.manifest.${{TEAMSFX_ENV}}.json"
+    ```
+
+    > [!NOTE]
+    > * Update the `manifestPath` value to the relative path of the Microsoft Entra app manifest template `aad.manifest.json`, if you've changed the file's path.
+    > * In a local setup, position the `aad/update` after the `file/createOrUpdateEnvironmentFile` action. This is because `aad/update` utilizes the output from `file/createOrUpdateEnvironmentFile`.
+
+* For React project, update `cli/runNpmCommand` under `deploy`:
+
+* If you're building a tab app using the React framework, find the `cli/runNpmCommand` action with `build app` in the `teamsapp.yml` file. Then, add the following environment variable:
+
+    ```yml
+    env:
+      REACT_APP_CLIENT_ID: ${{AAD_APP_CLIENT_ID}}
+      REACT_APP_START_LOGIN_PAGE_URL: ${{TAB_ENDPOINT}}/auth-start.html
+    ```
+
+* If you're building a tab app with React framework, find the `file/createOrUpdateEnvironmentFile` action for deploy in `teamsapp.local.yml` and add the following env:
+
+    ```yml
+    envs:
+      ...
+      REACT_APP_CLIENT_ID: ${{AAD_APP_CLIENT_ID}}
+      REACT_APP_START_LOGIN_PAGE_URL: ${{TAB_ENDPOINT}}/auth-start.html
+    ```
+
+## Update source code
+
+After implementing the above changes, your environment is prepared. You can now update your code to incorporate SSO into your Teams app.
+
+#### Vanilla JavaScript
+
+Use the following code to retrieve the SSO token and user information:
+
+```javascript
+function getSSOToken() {
+  return new Promise((resolve, reject) => {
+    microsoftTeams.authentication.getAuthToken()
+      .then((token) => resolve(token))
+      .catch((error) => reject("Error getting token: " + error));
+  });
+}
+
+function getBasicUserInfo() {
+  getSSOToken().then((ssoToken) => {
+    const tokenObj = JSON.parse(window.atob(ssoToken.split(".")[1]));
+    console.log(`username: ${tokenObj.name}`);
+    console.log(`user email: ${tokenObj.preferred_username}`);
+  });
+}
+```
+
+#### React
+
+For React projects, ensure the following environment variables are set in your deployment process:
+
+* For JavaScript project, see [tab JavaScript sample.](https://github.com/OfficeDev/teams-toolkit/tree/main/packages/fx-core/templates/plugins/resource/aad/auth/tab/js)
```

Comment:
The link seems doesn't related to environment variables mentioned above?

Reply:
@SLdragon, request you to please share the correct link.

Reply:
@SLdragon, request you to please share the correct link.